mod/messages/views/default/object/messages.php

```diff
@@ -18,7 +18,7 @@
 	if ($user) {
 		$icon = elgg_view_entity_icon($user, 'small');
 		$user_link = elgg_view('output/url', [
-			'href' => "messages/compose?send_to=$user->guid",
+			'href' => "messages/add?send_to=$user->guid",
```

Comment:
`elgg_generate_url()` should take care of these things. I don't think it makes any difference what actual paths are, so I would just leave it as compose.

Reply:
this probably makes it in line with all the rest of the urls used in elgg

Reply:
> this probably makes it in line with all the rest of the urls used in elgg

indeed, i like consistency

Reply:
Hm, we should probably allow multiple paths definitions, so that we don't unnecessarily break plugins, and this could also allow moving legacy URL logic in there.

```php
return [
    'routes' => [
        'add:object:message' => [
             'path' => [
                  '/messages/add',
                  '/messages/composer', // router will register this as 'add:object:message:aliasNth'
             ],
    ]
];
```